NEONLidarPipeline.py

```diff
@@ -0,0 +1,186 @@
+        """
+        import geopandas as gpd
+        plots_gdf = gpd.read_file(self.plot_path)
+        plots_gdf.geometry = plots_gdf.geometry.buffer(20)
+
+        return plots_gdf
+
+    def calculate_zonal_stats(self, plots_gdf):
+        """
+        Calculate zonal statistics (mean and max) for the given
+        GeoDataFrame using the lidar canopy height model (CHM).
+
+        Parameters
+        ----------
+        plots_gdf : geopandas.GeoDataFrame
+            A geopandas dataframe containing plot geometries.
+        Returns
+        -------
+        geopandas.GeoDataFrame
+            A geopandas dataframe containing the zonal statistics
+            (mean and max) for each plot.
+        """
+        import geopandas as gpd
+        import rasterstats as rs
+        site_chm_stats = rs.zonal_stats(
+            plots_gdf,
+            self.chm_path,
+            stats=['mean', 'max'],
+            geojson_out=True,
+            nodata=0,
+            copy_properties=True)
+
+        site_chm_stat_gdf = gpd.GeoDataFrame.from_features(site_chm_stats)
+        site_chm_stat_gdf.rename(
+            columns={'max': 'lidar_max', 'mean': 'lidar_mean'},
+            inplace=True)
+
+        return site_chm_stat_gdf
+
+    def get_insitu_data(self):
+        """
+        Reads the insitu vegetation structure data and calculates
+        the maximum and mean canopy height for each plot.
+
+        Returns
+        -------
+        geopandas.GeoDataFrame
+            A geopandas dataframe containing the maximum and mean
+            canopy height for each plot.
+        """
+        import pandas as pd
+        insitu_df = (pd.read_csv(self.insitu_path)
+                     [['plotid', 'stemheight']]
+                     .groupby('plotid')
+                     .stemheight
+                     .agg(['max', 'mean'])
+                     .rename(columns={'max': 'insitu_canopy_ht_max',
+                                      'mean': 'insitu_canopy_ht_mean'})
+                     .reset_index()
+                     )
+        insitu_df.rename(
+            columns={'stemheight': 'insitu_canopy_ht'},
+            inplace=True)
+
+        return insitu_df
+
+    def merge_insitu_data(self, site_name: str):
+        """
+        Merge the insitu data with the lidar-derived zonal statistics
+        data.
+
+        Parameters
+        ----------
+        site_name : str
+            The name of the site to process.
+            We need this to know how to merge the insitu data.
+            Not a general solution.
+
+        Returns
+        -------
+        geopandas.GeoDataFrame
+            A geopandas dataframe containing merged insitu and
+            lidar-derived zonal statistics data.
+        """
+        insitu_df = self.get_insitu_data()
+        site_chm_stat_gdf = self.execute_pipeline()
+
+        if site_name == 'sjer':
+            site_chm_stat_gdf = site_chm_stat_gdf.merge(
```

Comment:
@daltonkw I notice you haven't commented out many lines of this code - although many areas are pretty straightforward, it would be nice to have this part explained a bit more since it's so important.